CleverPush/CleverPush/Source/CleverPush.m

```diff
@@ -1958,6 +1958,60 @@ + (NSArray*)getNotifications {
     return notifications;
 }
 
+#pragma mark - Retrieving notifications based on the flag remote/local
++ (NSArray*)getNotifications:(BOOL)combineWithApi {
+    if (combineWithApi) {
+        NSString *combinedURL = [self createDynamicURLForNotification];
+        NSMutableArray* combinedArray = [[self getNotifications] mutableCopy];
+        NSMutableArray *remoteNotifications = [self getRemoteNotifications:combinedURL];
+        return [combinedArray arrayByAddingObjectsFromArray:remoteNotifications];
+    } else {
+        return [self getNotifications];
+    }
+}
+
+#pragma mark - Creating URL based on the topic dialogue and append the topicId's as a query parameter.
++ (NSString *)createDynamicURLForNotification {
+    NSString *baseURL = [NSString stringWithFormat:@"https://api.cleverpush.com/channel/%@/received-notifications?", channelId];
```

Comment:
Why do we need baseURL? The `path` in `[CleverPushHTTPClient sharedClient]` does not need domain, protocol, it only needs path.